nsfc-plugin/src/main/java/org/osc/controller/nsfc/utils/OsCalls.java

```diff
@@ -0,0 +1,269 @@
+        } catch (Exception e) {
+            throw new SdnControllerResponseNsfcException(Create, PortPair.class, e);
+        }
+
+        if (portPair == null) {
+            throw new RuntimeException("Create Port Pair operation returned null");
+        }
+
+        return portPair;
+    }
+
+    public List<? extends PortPairGroup> listPortPairGroups() {
+        return this.osClient.sfc().portpairgroups().list();
+    }
+
+    public List<? extends PortPair> listPortPairs() {
+        return this.osClient.sfc().portpairs().list();
+    }
+
+    public List<? extends PortChain> listPortChains() {
+        return this.osClient.sfc().portchains().list();
+    }
+
+    public FlowClassifier getFlowClassifier(String flowClassifierId) {
+        return this.osClient.sfc().flowclassifiers().get(flowClassifierId);
+    }
+
+    public PortChain getPortChain(String portChainId) {
+        PortChain portChain = this.osClient.sfc().portchains().get(portChainId);
+        return initializePortChainCollections(portChain);
+    }
+
+    public PortPairGroup getPortPairGroup(String portPairGroupId) {
+        return this.osClient.sfc().portpairgroups().get(portPairGroupId);
+    }
+
+    public PortPair getPortPair(String portPairId) {
+        return this.osClient.sfc().portpairs().get(portPairId);
+    }
+
+    public Port getPort(String portId) {
+        return this.osClient.networking().port().get(portId);
+    }
+
+    public PortChain updatePortChain(String portChainId, PortChain portChain) {
+        checkArgument(portChainId != null, "null passed for %s !", "Port Chain Id");
+        checkArgument(portChain != null, "null passed for %s !", "Port Chain");
+
+        // OS won't let us modify some attributes. Must be null on update object
+        portChain = portChain.toBuilder().id(null).projectId(null).chainParameters(null).chainId(null).build();
+
+        try {
+            portChain = this.osClient.sfc().portchains().update(portChainId, portChain);
+        } catch (Exception e) {
+            throw new SdnControllerResponseNsfcException(Update, PortChain.class, e);
+        }
+
+        if (portChain == null) {
+            throw new RuntimeException("Update Port Chain operation returned null");
+        }
+
+        return initializePortChainCollections(portChain);
+    }
+
+    public PortPairGroup updatePortPairGroup(String portPairGroupId, PortPairGroup portPairGroup) {
+        checkArgument(portPairGroupId != null, "null passed for %s !", "Port Pair Group Id");
+        checkArgument(portPairGroup != null, "null passed for %s !", "Port Pair Group");
+
+        // OS won't let us modify some attributes. Must be null on update object
+        portPairGroup  = portPairGroup.toBuilder().id(null).projectId(null).portPairGroupParameters(null).build();
+
+        try {
+            portPairGroup = this.osClient.sfc().portpairgroups().update(portPairGroupId, portPairGroup);
```

Comment:
can we use same pattern as below for consistency, throw runtime exception in the try block and rethrow as SdnControllerResponseNsfcException?